### PR TITLE
Add SectionNavigation component

### PR DESCRIPTION
**Pull request by gustaveen**
## Describe your changes
Add SectionNavigation component and remove preview of sections
- Consist of a back button and step indicator

Screenshot 2024-10-14 at 10 26 23

## Justify why they are needed
Updated design with new section navigation
## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by gustaveen** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4848?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4848** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4848?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4847** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4847?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gustaveen and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by guilhermespopolin — COMMENTED** [review]
Should the back button be hidden when scrolling over?

<div class='graphite__hidden'>
          <div>🎥 Video uploaded on Graphite:</div>
            <a href="https://app.graphite.dev/media/video/ocPE5OGCo5Y6fhJekeDQ/b178768c-f0cd-4c28-9b54-ecd370ab02f0.mov">
              
            </a>
          </div>
<video src="https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/ocPE5OGCo5Y6fhJekeDQ/b178768c-f0cd-4c28-9b54-ecd370ab02f0.mov">Screen Recording 2024-10-14 at 13.58.24.mov</video>

File: apps/store/src/features/priceCalculator/PurchaseFormV2/InsuranceDataForm/InsuranceDataForm.tsx [hunk 1]
```diff
@@ -49,13 +49,11 @@ export function InsuranceDataForm({ className }: { className?: string }) {
...
   const activeSectionId = useAtomValue(activeFormSectionIdAtom)
   const step = useAtomValue(priceCalculatorStepAtom)
+
   const sections = form.sections.map((section, index) => {
     if (step !== 'fillForm' || section.id !== activeSectionId) {
-      // No preview needed for sections that are not yet touched
-      if (section.state === 'initial') {
-        return null
-      }
-      return <SectionPreview key={section.id} section={section} />
```
Comment (alebedev) [inline apps/store/src/features/priceCalculator/PurchaseFormV2/InsuranceDataForm/InsuranceDataForm.tsx:58]: Can we delete `SectionPreview` now?

File: apps/store/src/features/priceCalculator/PurchaseFormV2/InsuranceDataForm/SectionNavigation.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  priceCalculatorStepAtom,
+} from '../../priceCalculatorAtoms'
+
+export function SectionNavigation() {
+  const { t } = useTranslation('purchase-form')
+  const form = useAtomValue(priceCalculatorFormAtom)
+  const activeSectionId = useAtomValue(activeFormSectionIdAtom)
+  const setShowEditSsnWarning = useSetAtom(priceCalculatorShowEditSsnWarningAtom)
+  const setStep = useSetAtom(priceCalculatorStepAtom)
+  const setActiveSectionId = useSetAtom(activeFormSectionIdAtom)
```
Comment (alebedev) [inline apps/store/src/features/priceCalculator/PurchaseFormV2/InsuranceDataForm/SectionNavigation.tsx:21]: Can you combine this and `useAtomValue(activeFormSectionIdAtom)` into `useAtom()`?

**Review by alebedev — APPROVED** [review]
Looks good! Few comments

File: apps/store/src/features/priceCalculator/PurchaseFormV2/InsuranceDataForm/InsuranceDataForm.tsx (hunk 1 above)
Comment (gustaveen) [reply apps/store/src/features/priceCalculator/PurchaseFormV2/InsuranceDataForm/InsuranceDataForm.tsx:58]: Didn't want to remove it in this PR if we would feel we should go live with the editable sections instead. But sure, we can always go back in history 👍

File: apps/store/src/features/priceCalculator/PurchaseFormV2/InsuranceDataForm/SectionNavigation.tsx (hunk 2 above)
Comment (gustaveen) [reply apps/store/src/features/priceCalculator/PurchaseFormV2/InsuranceDataForm/SectionNavigation.tsx:21]: Hm, yeah might be better to refactor. Let's do it in a new PR so we can merge this for the test session

**Review by guilhermespopolin — APPROVED** [review]

**Comment by gustaveen** [comment]
### Merge activity

* **Oct 14, 9:39 AM EDT**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4848).

**Pull request merged**